### PR TITLE
fix: rename list bucket/object reponse class

### PR DESCRIPTION
**Pull request by macnev2013**
# Issue
- #7541 

# Fix
- Patched `ListBucketsOutput` to `ListAllMyBucketsResult`.
  - In boto spec definition of output from `ListBuckets` is `ListBucketsOutput` whereas AWS is returning  `ListAllMyBucketsResult`.
- `list_objects` contains `ListBucketResult` object whereas we are returning `ListObjectsOutput` 

# Commands to Reproduce

```
Start localstack instance with env `PROVIDER_OVERRIDE_S3: "asf"`
```

Comparing the response from this commands will po
`list_bucket`
```
awslocal s3api list-buckets --profile ls --debug
```
```
aws s3api list-buckets --profile ls --debug
```

`list-objects`
```
awslocal s3 mb s3://test-bucket --profile ls --region us-east-1 --debug
awslocal s3api list-objects --bucket test-bucket --profile ls --region us-east-1 --debug
```
```
aws s3 mb s3://test-bucket --profile ls --region us-east-1 --debug
aws s3api list-objects --bucket test-bucket --profile ls --region us-east-1 --debug
```

# Tested Also with
https://github.com/Dowwie/localstack_s3_xml_issue

### CODE REVIEW

File: localstack/aws/spec-patches.json
Comment (thrau) [inline localstack/aws/spec-patches.json:799]: just asking (haven't checked): could this be simplified with the "move", or "replace" operation? https://jsonpatch.com/#move

**Review by thrau — APPROVED** [review]
LGTM! we should double check if the tests pass with `PROVIDER_OVERRIDE_S3=asf`. and also i'll let @bentsku do the final sign-off

File: localstack/aws/spec-patches.json
Comment (macnev2013) [reply localstack/aws/spec-patches.json]: Hey @thrau, Move worked in this case. thanks for the suggestion :)

**Comment by macnev2013** [comment]
> LGTM! we should double check if the tests pass with `PROVIDER_OVERRIDE_S3=asf`. and also i'll let @bentsku do the final sign-off

It was working with the older provider. The fix is mainly focused on `PROVIDER_OVERRIDE_S3=asf`. (I've double checked it  though 👍🏻 )

**Comment by alexrashed** [comment]
> LGTM! we should double check if the tests pass with PROVIDER_OVERRIDE_S3=asf. and also i'll let @bentsku do the final sign-off
@thrau We added the S3 ASF provider to the test pipeline some time ago, in order to make sure that there are no regressions introduced along the way. You can see the run for this PR here: https://app.circleci.com/pipelines/github/localstack/localstack/12048/workflows/9d86dafc-9ad2-4547-aa14-2cfec3162855/jobs/85316

However, these tests aren't reliable in this case, because they weren't failing before this bug was fixed.
This isn't really surprising, because the root elements of the XML responses are often ignored by clients / response parsers.
So I would actually be in favor of a specific test for this fix before merging it. It could be a simple request to the API which then accesses the response content, parses the XML response, and checks that the root tag is in fact the correct one.

**Review by bentsku — CHANGES_REQUESTED** [review]
I share Alex's opinion, we should write a simple test targeting the S3 endpoint for AWS and LS and test the content of the response (could use `xmltodict`), maybe using pre-signed URL for managing authentication of the request (making it easier to parse the response). Then we should be good to go! Thanks for tackling this and getting to know specs patches! 😄

**Review by bentsku — APPROVED** [review]
LGTM! 🚀 thanks for adding the test!

**Comment by Dowwie** [comment]
@thrau

**Comment by macnev2013** [comment]
Hi @Dowwie, Apologies for delayed response. You can remove the example. Thanks for your help 🚀

**Pull request merged**
